lib/openapi_first/definition.rb

```diff
@@ -40,6 +40,24 @@ def initialize(contents, filepath = nil)
     # @return [Enumerable[Router::Route]]
     def_delegators :@router, :routes
 
+    # Returns a unique identifier for this API definition
+    # @return [String] A unique key for this API definition
+    def key
+      return filepath if filepath
+
+      info = self['info'] || {}
+      title = info['title']
+      version = info['version']
+
+      if title.nil? || version.nil?
+        raise ArgumentError,
+              "Cannot generate key for the OpenAPI document because 'info.title' or 'info.version' is missing. " \
+              "Please add these fields to your OpenAPI document."
+      end
+
+      "#{title} @ #{version}"
```

Comment:
Thoughts? You mentioned using a hash of the definition. Would you prefer that over this?

Reply:
Until now openapi_first did not have an opinion about the contents of the OAD…, but I think its' fine the way you did it, because it will lead to a human readable key.